hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -991,17 +993,19 @@ class HoodieSparkSqlWriterInternal {
                                              tableInstantInfo: TableInstantInfo,
                                              extraPreCommitFn: Option[BiConsumer[HoodieTableMetaClient, HoodieCommitMetadata]]
                                             ): (Boolean, HOption[java.lang.String], HOption[java.lang.String]) = {
-    if (writeResult.getWriteStatuses.rdd.filter(ws => ws.hasErrors).count() == 0) {
-      log.info("Proceeding to commit the write.")
-      val metaMap = parameters.filter(kv =>
-        kv._1.startsWith(parameters(COMMIT_METADATA_KEYPREFIX.key)))
-      val commitSuccess =
-        client.commit(tableInstantInfo.instantTime, writeResult.getWriteStatuses,
-          common.util.Option.of(new java.util.HashMap[String, String](metaMap.asJava)),
-          tableInstantInfo.commitActionType,
-          writeResult.getPartitionToReplaceFileIds,
-          common.util.Option.ofNullable(extraPreCommitFn.orNull))
-
+    val hasErrors = new AtomicBoolean(false)
```

Comment:
Note to Reviewer: 
No major changes in L996 to 1008, except adding the SparkDataSourceWriteStatusHandlerCallback.